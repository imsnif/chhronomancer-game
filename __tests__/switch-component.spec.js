import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))

test('Switch (two options) => renders correctly', () => {
  const Switch = require('../components/switch').default
  const tree = renderer.create(
    <Switch selected={0} options={[
      {action: () => {}, text: 'foo'},
      {action: () => {}, text: 'bar'}
    ]} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Switch (three options) => renders correctly', () => {
  const Switch = require('../components/switch').default
  const tree = renderer.create(
    <Switch selected={2} options={[
      {action: () => {}, text: 'foo'},
      {action: () => {}, text: 'bar'},
      {action: () => {}, text: 'baz'}
    ]} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
