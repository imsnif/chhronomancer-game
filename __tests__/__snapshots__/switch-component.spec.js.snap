// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch (three options) => renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#329932",
      "borderLeftWidth": 1,
      "borderRightWidth": 1,
      "borderTopWidth": 1,
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "backgroundColor": "#141313",
          "borderBottomColor": "#329932",
          "borderBottomWidth": 1,
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#329932",
          "fontFamily": "telegrama_raw",
          "fontSize": 45.911331934501746,
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
    >
      foo
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "backgroundColor": "#141313",
          "borderBottomColor": "#329932",
          "borderBottomWidth": 1,
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#329932",
          "fontFamily": "telegrama_raw",
          "fontSize": 45.911331934501746,
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
    >
      bar
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "backgroundColor": "#329932",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#0e0e0e",
          "fontFamily": "telegrama_raw",
          "fontSize": 45.911331934501746,
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
    >
      baz
    </Text>
  </View>
</View>
`;

exports[`Switch (two options) => renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#329932",
      "borderLeftWidth": 1,
      "borderRightWidth": 1,
      "borderTopWidth": 1,
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "backgroundColor": "#329932",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#0e0e0e",
          "fontFamily": "telegrama_raw",
          "fontSize": 45.911331934501746,
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
    >
      foo
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "backgroundColor": "#141313",
          "borderBottomColor": "#329932",
          "borderBottomWidth": 1,
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#329932",
          "fontFamily": "telegrama_raw",
          "fontSize": 45.911331934501746,
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
    >
      bar
    </Text>
  </View>
</View>
`;
